src/models/encrypted-model.ts

```diff
@@ -0,0 +1,30 @@
+import { Model } from "sequelize";
+import { EncryptionClient, CryptorSecretKey } from "../util/encryption-client";
+import Logger from "bunyan";
+
+export abstract class EncryptedModel<TModel extends EncryptedModel<TModel, TSecretField>, TSecretField extends keyof TModel> extends Model {
+
+	abstract getEncryptContext(field: TSecretField): Promise<Record<string, string | number>>;
+	abstract getCryptorKeyAlias(): CryptorSecretKey;
+	abstract getAllSecretFields(): TSecretField[];
+
+	async decrypt(field: TSecretField, logger?: Logger) {
+		const ctx = await this.getEncryptContext(field);
+		return await EncryptionClient.decrypt(this.getDataValue(field as any), ctx, logger);
+	}
+
+	protected async encrypt(field: TSecretField, logger?: Logger) {
```

Comment:
Didn't open it as:
1. We don't need it outside,
2. Avoid double-encryption accidentally by consumer. (Like `app.secrete = await EncryptClient.encrypt('plain-text')` )